addons/assignGear/functions/fnc_assignGearSupplyBox.sqf

```diff
@@ -40,46 +41,35 @@ if (!isClass _path) exitWith {
     diag_log text format ["[POTATO-assignGear] - No loadout found for %1 (typeOf %2)", _theBox, typeOf _theBox];
 };
 
-//Clean out starting inventory (even if there is no class)
-clearWeaponCargoGlobal _theBox;
-clearMagazineCargoGlobal _theBox;
-clearItemCargoGlobal _theBox;
-clearBackpackCargoGlobal _theBox;
-
-private _transportMagazines = getArray(_path >> "TransportMagazines");
-private _transportItems = getArray(_path >> "TransportItems");
-private _transportWeapons = getArray(_path >> "TransportWeapons");
-private _transportBackpacks = getArray(_path >> "TransportBackpacks");
-
-// transportMagazines
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    _theBox addMagazineCargoGlobal [_classname, parseNumber _amount];
-    nil
-} count _transportMagazines; // count used here for speed, make sure nil is above this line
-
-// transportItems
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    _theBox addItemCargoGlobal [_classname, parseNumber _amount];
-    nil
-} count _transportItems; // count used here for speed, make sure nil is above this line
-
-// transportWeapons
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    private _disposableName = [cba_disposable_LoadedLaunchers, _classname, "get", ""] call FUNC(getDisposableInfo);
-    if (_disposableName != "") then {
-        TRACE_2("cba_disposable_LoadedLaunchers replace",_classname,_disposableName);
-        _classname = _disposableName;
+private _subBoxes = "true" configClasses _path;
+if (_subBoxes isNotEqualTo []) then {
+    private _boxName = getText (_path >> "boxCustomName");
+    if (_boxName isNotEqualTo "") then {
+        _theBox setVariable ["ace_cargo_customName", _boxName, true];
     };
-    _theBox addWeaponCargoGlobal [_classname, parseNumber _amount];
-    nil
-} count _transportWeapons; // count used here for speed, make sure nil is above this line
-
-// transportBackpacks
-{
-    (_x splitString ":") params ["_classname", ["_amount", "1", [""]]];
-    _theBox addBackpackCargoGlobal [_classname, parseNumber _amount];
-    nil
-} count _transportBackpacks; // count used here for speed, make sure nil is above this line
+    clearWeaponCargoGlobal _theBox;
+    clearMagazineCargoGlobal _theBox;
+    clearItemCargoGlobal _theBox;
+    clearBackpackCargoGlobal _theBox;
+    private _boxSpace = getNumber (_path >> "boxSpace");
+    [_theBox, [_boxSpace, 4] select (_boxSpace == 0)] call ace_cargo_fnc_setSpace;
+    {
+        private _subBoxType = configName _x;
+        private _boxCount = (getNumber (_x >> "boxCount")) max 1;
+        for "_i" from 1 to _boxCount do {
+            private _subBox = createVehicle [_subBoxType, [23,54,975], [], 0, "CAN_COLLIDE"];
```

Comment:
why this specific position?

Reply:
Old habit to not use [0, 0, 0] because sometimes weird stuff can be there simultaneously. It can definitely just be [0, 0, 0]

Reply:
changed to [0, 0, 0]